gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerAuthKrb5IntegrateTest.java

```diff
@@ -134,7 +133,7 @@ public void shouldFailWithoutClientJaasEntry() throws Exception {
             fail("This should not succeed as the client config does not contain a JaasEntry");
         } catch(Exception ex) {
             final Throwable root = ExceptionUtils.getRootCause(ex);
-            assertTrue(root instanceof ResponseException || root instanceof GSSException);
+            assertTrue(root instanceof TimeoutException);
```

Comment:
two things:

1. The reason for checking both `ResponseException` and `GSSException` is that in different environments (mac os, linux, travis, etc) we tend to see  different exceptions thrown. I've never known why that is except that we used to have a ton of unexpected test failures before we started checking both. Ever since then, Gremlin Server integration tests have been very solid. Did you do something here to ensure that we always get `TimeoutException`.
2. `TimeoutException` doesn't seem like the right exception to be thrown here (or in the test below). We have a bad client configuration - seems like any one of the other exceptions that were previously thrown were better than `TimeoutException` - thoughts?

Reply:
I reverted the code to check both `ResponseException` and `TimeoutException`

Reply:
and replaced `TimeoutException` with a dedicated `NoHostAvailableException` when no host is capable to serve the request